src/main/java/uk/gov/hmcts/ccd/data/casedetails/DefaultCaseDetailsRepository.java

```diff
@@ -157,6 +174,27 @@ public PaginatedSearchMetadata getPaginatedSearchMetadata(MetaData metaData, Map
         return sr;
     }
 
+    // TODO This accepts null values for backward compatibility. Once deprecated methods are removed, parameters should
+    // be annotated with @NotNull
+    private Optional<CaseDetailsEntity> find(String jurisdiction, Long id, String reference) {
+        final CaseDetailsQueryBuilder qb = queryBuilderFactory.select(em);
+
+        if (null != jurisdiction) {
+            qb.whereJurisdiction(jurisdiction);
```

Comment:
not sure whether would be better, but we could have whereJurisdiction method do the null check itself. 
This way we avoid repeated null checks in the clients of query builder

Reply:
The only instances where `jurisdiction` may be `null` is where deprecated methods are used. By keeping that check close to the deprecated code, it will be easier to clean. Moreover, `null` jurisdiction is not the expected behaviour so I wouldn't want to facilitate it.